src/catalog/database_catalog.cpp

```diff
@@ -334,14 +334,19 @@ void DatabaseCatalog::BootstrapIndex(const common::ManagedPointer<transaction::T
 bool DatabaseCatalog::CreateTableEntry(const common::ManagedPointer<transaction::TransactionContext> txn,
                                        const table_oid_t table_oid, const namespace_oid_t ns_oid,
                                        const std::string &name, const Schema &schema) {
+  if (pg_core_.CreateTableEntry(txn, table_oid, ns_oid, name, schema)) {
+    CreateTableStatisticEntry(txn, table_oid, schema);
+    return true;
+  }
+  return false;
+}
+
+void DatabaseCatalog::CreateTableStatisticEntry(const common::ManagedPointer<transaction::TransactionContext> txn,
+                                                const table_oid_t table_oid, const Schema &schema) {
   // Create associated entries in pg_statistic.
-  {
-    col_oid_t col_oid(1);
```

Comment:
So I looked into it. The col oid for every column in `schema` is 0. That's because the table hasn't been created yet and col oids haven't been assigned yet. Even with your change of moving `pg_stat_.CreateColumnStatistic` to after the table creation, every col oid in `schema` is still 0. My guess is that the schema object doesn't actually get updated when you create a table and you'll have to pull a new one with the actual col oids.

You'll need to add something like `const auto &updated_schema = GetSchema(txn, table_oid)` and pull col oids from that.

Reply:
```suggestion
  const auto &updated_schema = GetSchema(txn, table_oid)
  for (const auto &col : updated_schema.GetColumns()) {
```